### PR TITLE
Add microphone input warning notification

### PR DESCRIPTION
**Pull request by ItsSimko**
## Summary by Sourcery

Add a feature to notify users of potential microphone input issues by displaying a warning after 10 seconds of silence. Enhance the UI with a warning bar to provide visual feedback and allow users to close the notification.

New Features:
- Introduce a warning notification for microphone input issues, alerting users after 10 seconds of silence.

Enhancements:
- Add a warning bar to the UI to notify users about microphone issues, with options to display and close the warning.

image

### CODE REVIEW

**Comment by ItsSimko** [comment]
@sourcery-ai review

File: src/FreeScribe.client/client.py [hunk 1]
```diff
@@ -178,6 +178,7 @@
         elif current_view == "minimal":
             pause_button.config(text="⏸️", bg=DEFAULT_BUTTON_COLOUR)
     
+SILENCE_WARNING_LENGTH = 10 # seconds, warn the user after 10s of no input something might be wrong
 
 def record_audio():
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/client.py:183]: https://github.com/ClinicianFOCUS/FreeScribe/pull/219/commits/bed8420d1b2f9d350403ce5ad3e3eec5a5498316

Moved new stuff to new function to hopefully improve a bit

File: src/FreeScribe.client/client.py [hunk 2]
```diff
@@ -209,12 +211,25 @@
...
             
             record_duration += CHUNK / RATE
-            
+
+            # Check if we need to warn if silence is long than warn time
+            if silent_warning_duration >= SILENCE_WARNING_LENGTH:
+                
+                # If the warning bar is not already displayed, create it
+                if window.warning_bar is None:
+                    window.create_warning_bar("No audio input detected for 10 seconds. Please check your microphone input device in whisper settings and adjust your microphone cutoff level in advanced settings.")
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/client.py]: https://github.com/ClinicianFOCUS/FreeScribe/pull/219/commits/368d853c95fb7b73bc51a8d0c6b38483601f3c00

File: src/FreeScribe.client/client.py [hunk 3]
```diff
@@ -231,6 +246,9 @@ def record_audio():
     stream.close()
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/client.py]: https://github.com/ClinicianFOCUS/FreeScribe/pull/219/commits/4a4bad187f20ecbda903b0e2058244677ed80441

File: src/FreeScribe.client/UI/MainWindowUI.py [hunk 4]
```diff
@@ -123,6 +125,51 @@ def create_docker_status_bar(self):
         self._background_availbility_docker_check()
         self._background_check_container_status(llm_dot, whisper_dot)
 
+    def create_warning_bar(self, text, row=3, column=0, columnspan=11, pady=10, padx=10, sticky='nsew'):
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/UI/MainWindowUI.py]: https://github.com/ClinicianFOCUS/FreeScribe/pull/219/commits/368d853c95fb7b73bc51a8d0c6b38483601f3c00

File: src/FreeScribe.client/client.py [hunk 5]
```diff
@@ -178,6 +178,7 @@ def toggle_pause():
         elif current_view == "minimal":
             pause_button.config(text="⏸️", bg=DEFAULT_BUTTON_COLOUR)
     
+SILENCE_WARNING_LENGTH = 10 # seconds, warn the user after 10s of no input something might be wrong
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/FreeScribe.client/client.py:181]: This is good - Alex can you change?

[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/client.py:181]: Yes working on that now. Just implemented the bug fix.

**Review by yingbull — CHANGES_REQUESTED** [review]
Add the use of the 10s constant in the message

File: src/FreeScribe.client/client.py [hunk 6]
```diff
@@ -194,43 +195,72 @@
...
-    # Send any remaining audio chunk when recording stops
-    if current_chunk:
-        audio_queue.put(b''.join(current_chunk))
-
-    stream.stop_stream()
-    stream.close()
-    audio_queue.put(None)
+    try:
 
+        current_chunk = []
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/client.py:200]: Ill leave this for later. Please create a issue ticket. this is legacy code

**Comment by ItsSimko** [comment]
Completed sourcery suggestions ready for merge

**Review by yingbull — APPROVED** [review]

**Pull request merged**